packages/@aws-cdk/aws-vpcv2-alpha/lib/ipam.ts

```diff
@@ -0,0 +1,428 @@
+ * @default BYOIP
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipampool.html#cfn-ec2-ipampool-publicipsource
+ */
+export enum IpamPoolPublicIpSource {
+  /**
+   * BYOIP Ipv6 to be registered under IPAM
+   */
+  BYOIP = 'byoip',
+
+  /**
+   * Amazon Provided Ipv6 range
+   */
+  AMAZON = 'amazon',
+}
+
+/**
+ * Options to provisioned CIDRs to an IPAM pool.
+ * Used to create a new IpamPoolCidr
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipampoolcidr.html
+ */
+export interface IpamPoolCidrProvisioningOptions {
+  /**
+   * Ipv6 Netmask length for the CIDR
+   * @default none
+   */
+  readonly netmaskLength?: number;
+
+  /**
+   * Ipv6 CIDR block for the IPAM pool
+   * @default none
+   */
+  readonly cidr?: string;
+}
+
+/**
+ * Creates new IPAM Pool
+ * Pools enable you to organize your IP addresses according to your routing and security needs
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipampool.html
+ * @resource AWS::EC2::IPAMPool
+ */
+export class IpamPool extends Resource {
+
+  /**
+   * Pool ID to be passed to the VPC construct
+   * @attribute IpamPoolId
+   */
+  public readonly ipamPoolId: string;
+
+  /**
+   * Pool CIDR for IPv6 to be provisioned with Public IP source set to 'Amazon'
+   */
+  public readonly ipamCidrs: IpamPoolCidr[] = []
+
+  constructor(scope: Construct, id: string, props: IpamPoolProps) {
+    super(scope, id, {
+      physicalName: props.ipamPoolName ?? Lazy.string({
+        produce: () => Names.uniqueResourceName(this, { maxLength: 128, allowedSpecialCharacters: '_' }),
+      }),
+    });
+    const cfnPool = new CfnIPAMPool(this, id, {
+      addressFamily: props.addressFamily,
+      provisionedCidrs: props.provisionedCidrs,
+      locale: props.locale,
+      ipamScopeId: props.ipamScopeId,
+      publicIpSource: props.publicIpSource,
+      awsService: props.awsService,
+    });
+    this.ipamPoolId = cfnPool.attrIpamPoolId;
+  }
+
+  /**
+   * A CIDR provisioned to an IPAM pool.
+   * @param id Name of Resource
+   * @param options Either a CIDR or netmask length must be provided
+   * @returns AWS::EC2::IPAMPoolCidr
+   */
+  public provisionCidr(id: string, options: IpamPoolCidrProvisioningOptions): IpamPoolCidr {
+    const cidr = new IpamPoolCidr(this, id, {
+      ipamPoolCidrName: id,
+      ...options,
+      ipamPoolId: this.ipamPoolId,
+    });
+    this.ipamCidrs.push(cidr);
+    return cidr;
+  }
+}
+
+/**
+ * Properties for defining a CIDR range within an IPAM pool.
+ *
+ * @extends IpamPoolCidrProvisioningOptions - Base options for provisioning a CIDR range.
+ *
+ * @property {string} ipamPoolId - The ID of the IPAM pool where the CIDR range will be provisioned.
+ */
+export interface IpamPoolCidrProps extends IpamPoolCidrProvisioningOptions {
+  /**
+   * Ipam Pool ID to add provisioned CIDR
+   */
+  readonly ipamPoolId: string;
```

Comment:
this is not an input by the user but something that we are passing from the implementation of below method, for user it will be calling the function on pool object itself

pool2.provisionCidr('PublicPool0Cidr', { netmaskLength: 52 } );